File: tests/fixtures/java_basic/src/main/java/acme/PaymentService.java
package acme;

import java.util.Objects;

public class PaymentService {
  private final PaymentGateway gateway;

  public PaymentService(PaymentGateway gateway) {
    this.gateway = gateway;
  }

  public boolean authorize(PaymentRequest request) {
    validate(request);
    return gateway.charge(request);
  }

  private void validate(PaymentRequest request) {
    if (request == null) {
      throw new IllegalArgumentException("request");
    }
    Objects.requireNonNull(request.userId(), "userId");
  }
}

File: tests/fixtures/cpp_basic/src/payment_service.cpp
#include "acme/payment_service.hpp"

#include <stdexcept>

namespace acme {

PaymentService::PaymentService(PaymentGateway* gateway) : gateway_(gateway) {}

bool PaymentService::authorize(const PaymentRequest& request) {
  validate(request);
  return gateway_->charge(request);
}

bool PaymentService::validate(const PaymentRequest& request) {
  if (request.amount <= 0) {
    throw std::invalid_argument("amount");
  }
  return true;
}

}  // namespace acme
